src/software/new_geom/linear_spline2d.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <vector>
+
+#include "software/new_geom/point.h"
+#include "software/new_geom/polynomial2d.h"
+#include "software/new_geom/spline2d.h"
+
+/**
+ * A linear spline is a spline composed of a series of linear 2d polynomials, ie. a series
+ * of connected line segments
+ */
+class LinearSpline2d : Spline2d
```

Comment:
Just curious, why not call this `FirstOrderSpline2d` or similar? Just asking because I expect that to be the naming scheme of higher-order splines, so might be worth being consistent.

Reply:
I think `Linear`, `Quadratic`, `Cubic`, etc. is a more intuitive naming scheme for most people.

Reply:
ah right those are also names. SGTM